hw/ip/hmac/doc/registers.md

```diff
@@ -184,7 +184,13 @@ If the software updates the register while the engine computes the hash, the upd
 Key length configuration.
 
 This is a 6-bit one-hot encoded field to configure the key length for HMAC.
-The HMAC supports key lengths of 128-bit, 256-bit, 384-bit, 512-bit and 1024-bit, as long as the key length is not greater than the block size: up to 1024-bit for SHA-2 384/512 and up to 512-bit for SHA-2 256.
+
+The HMAC can be programmed with the following key lengths: 128-bit, 256-bit, 384-bit, 512-bit and 1024-bit.
+But the HMAC supports any arbitrary key length, to do so the software should configure the HMAC to the next largest supported key length and concatenate zeros to reach the programmed key length.
```

Comment:
But the HMAC supports any arbitrary key length: the software should configure the HMAC with the next largest supported key length and concatenate zeros to reach the programmed key length.